### PR TITLE
Better sheath

### PR DESCRIPTION
**Pull request by actioninja**
## About The Pull Request

Makes the captain's saber sheath less shitty.
Click to remove the sword, click the sheath with the sword to put it back. Click drag it to hand to get it out of slots, you know the drill.
It's still a one slot storage so all hotkeys still work.

Also fixed an absolutely ancient bug where the sheath wasn't playing sound effects for drawing and removing the sword

## Why It's Good For The Game

less clunk
fix bug

## Changelog
:cl:
tweak: streamlined the captain's saber sheath
fix: captain's saber sheath plays sounds like it should
/:cl:

### CODE REVIEW

**Comment by Shapsy** [comment]
That's a pretty massive buff to let you carry around the saber (one of the most dangerous melee weapons in the game) for free. The sheath isn't there to let you store the sword more efficiently, it's there to let you store it at all

**Comment by MrDoomBringer** [comment]
@Shapsy ????

**Comment by granpawalton** [comment]
Rip using shift e on it

**Comment by PKPenguin321** [comment]
You could already draw it quickly with alt click

**Comment by py01** [comment]
Does this prevent you from using shift+e to quick draw/sheath it? It isn't clear whether or not it stops that

**Comment by granpawalton** [comment]
actually im being dumb you can just store the sword itself on the belt even if you cant store the sheath so it doesnt hurt anyone wanting to use that

**Comment by ATH1909** [comment]
If this doesn't already have shift+e functionality, it should be given some.

File: code/game/objects/items/storage/sheath.dm [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+		if(!iscarbon(user) || !user.canUseTopic(src, BE_CLOSE, ismonkey(user)))
+			return
+		if(holding)
+			user.visible_message("<span class='notice'>[user] takes \the [holding] out of \the [src].</span>", "<span class='notice'>You take [holding] out of [src].</span>")
+			user.put_in_hands(holding)
+			holding = null
+			playsound(src, 'sound/items/unsheath.ogg', 25, TRUE)
+			update_icon()
+			return
+	. = ..()
```
Comment (ShizCalev) [inline code/game/objects/items/storage/sheath.dm]: return ..()

File: code/game/objects/items/storage/sheath.dm [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	if(!istype(I, hold_type))
+		to_chat(user, "<span class='warning'>\The [I] doesn't seem to fit into \the [src]...</span>")
+		return
+	if(user.transferItemToLoc(I, src))
+		holding = I
+		playsound(src, 'sound/items/sheath.ogg', 25, TRUE)
+		user.visible_message("<span class='notice'>[user] puts \the [I] into \the [src].</span>", "<span class='notice'>You take \the [holding] out of \the [src].</span>")
+		update_icon()
+		return
+	. = ..()
```
Comment (ShizCalev) [inline code/game/objects/items/storage/sheath.dm]: .

File: code/game/objects/items/storage/sheath.dm [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+/obj/item/sheath/MouseDrop(atom/over, src_location, over_location, src_control, over_control, params)
+	if(istype(over, /obj/screen/inventory/hand))
+		var/obj/screen/inventory/hand/H = over
+		var/mob/M = usr
+		M.putItemFromInventoryInHandIfPossible(src, H.held_index)
+		return
+	. = ..()
```
Comment (ShizCalev) [inline code/game/objects/items/storage/sheath.dm]: .

File: code/game/objects/items/storage/sheath.dm [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+	if(!holding)
+		icon_state = initial(icon_state)
+		item_state = initial(icon_state)
+	else
+		icon_state = "[initial(icon_state)]-sabre"
+		item_state = "[initial(icon_state)]-sabre"
+	if(loc && isliving(loc))
+		var/mob/living/L = loc
+		L.regenerate_icons()
+	..()
```
Comment (ShizCalev) [inline code/game/objects/items/storage/sheath.dm]: .

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by kriskog** [comment]
I think the quickdraw with shift + e comments should be taken into consideration. Is there any reason to keep the sheath now, and not just put the sword directly in belt for the shift + e quickdraw?

**Comment by actioninja** [comment]
@ShizCalev alright this is ready now.

File: code/game/objects/items/melee/misc.dm [hunk 5]
```diff
@@ -76,17 +76,11 @@
 		final_block_chance = 0 //Don't bring a sword to a gunfight
 	return ..()
 
-/obj/item/melee/sabre/on_exit_storage(obj/item/storage/S)
-	..()
-	var/obj/item/storage/belt/sabre/B = S
-	if(istype(B))
-		playsound(B, 'sound/items/unsheath.ogg', 25, 1)
```
Comment (ShizCalev) [inline code/game/objects/items/melee/misc.dm]:
```
/obj/item/melee/sabre/on_exit_storage(datum/component/storage/concrete/S)
	..()
	var/obj/item/storage/belt/sabre/B = S.real_location()
	if(istype(B))
		playsound(B, 'sound/items/unsheath.ogg', 25, 1)
```

File: code/game/objects/items/melee/misc.dm [hunk 6]
```diff
@@ -76,17 +76,11 @@
...
+/obj/item/melee/sabre/on_exit_storage(datum/component/storage/concrete/S)
+	playsound(src, 'sound/items/unsheath.ogg', 25, 1)
 
-/obj/item/melee/sabre/on_enter_storage(obj/item/storage/S)
-	..()
-	var/obj/item/storage/belt/sabre/B = S
-	if(istype(B))
-		playsound(B, 'sound/items/sheath.ogg', 25, 1)
```
Comment (ShizCalev) [inline code/game/objects/items/melee/misc.dm]: .

File: code/game/objects/items/storage/belt.dm [hunk 7]
```diff
@@ -636,50 +636,63 @@
...
-	name = "sabre sheath"
-	desc = "An ornate sheath designed to hold an officer's blade."
+/obj/item/storage/belt/sheath
 	icon_state = "sheath"
 	item_state = "sheath"
 	w_class = WEIGHT_CLASS_BULKY
+	var/hold_type
 
-/obj/item/storage/belt/sabre/ComponentInitialize()
+/obj/item/storage/belt/sheath/Initialize()
```
Comment (ShizCalev) [inline code/game/objects/items/storage/belt.dm]: this doesn't change anything, set it back to ComponentInitialize() (it's called at the end of Initialize anyway)

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/game/objects/items/storage/belt.dm (hunk 7 above)
Comment (actioninja) [reply code/game/objects/items/storage/belt.dm]: this was just an accident, thank you code review

File: code/game/objects/items/melee/misc.dm (hunk 5 above)
Comment (actioninja) [reply code/game/objects/items/melee/misc.dm]: I deleted the parent call because the parent call was just a stub.

**Comment by optimumtact** [comment]
How is this better?

**Comment by actioninja** [comment]
It's easier to use.

**Comment by optimumtact** [comment]
like soon? or did you want this merged

**Comment by actioninja** [comment]
I'm just going to close this until I get around to doing that. The sound fix will be split off real quick.

**Pull request closed without merging**